jetty-server/src/main/java/org/eclipse/jetty/server/ServletPathMapping.java

```diff
@@ -0,0 +1,148 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.server;
+
+import javax.servlet.http.HttpServletMapping;
+import javax.servlet.http.MappingMatch;
+
+import org.eclipse.jetty.http.pathmap.PathSpec;
+import org.eclipse.jetty.http.pathmap.ServletPathSpec;
+
+public class ServletPathMapping implements HttpServletMapping
+{
+    private final MappingMatch _mappingMatch;
+    private final String _matchValue;
+    private final String _pattern;
+    private final String _servletName;
+    private final String _servletPath;
+    private final String _pathInfo;
+
+    public ServletPathMapping(PathSpec pathSpec, String servletName, String pathInContext)
```

Comment:
Need a comment in here about the purpose of this class, and particularly how it is used to model both mapping via a path, but ALSO mapping via just a name. In fact, I don't like the name of this class containing the word "path" if path has nothing to do with it.

Reply:
I would call it just `ServletMapping`, but that name is already taken!  Let me see if we can avoid using it at all for named dispatch....

Reply:
While you're at it, the ServletMapping class could do with a bit of an overhaul:  the setDefault()/isDefaul() method names I now see are confusing with the whole other meaning of "default". In the case of the ServletMapping these methods just mean whether or not the mapping originated from a webdefault.xml. Can you think of a clearer name?

Reply:
It turned out that named dispatchers were broken. Added a test for them and now no longer use a ServletPathMapping for named dispatcher.  Added some javadoc as well.

Reply:
rename of isDefault done.